vpc.tf

```
@@ -0,0 +1,109 @@
+# VPC
+resource "aws_vpc" "task1_vpc" {
+  cidr_block       = var.vpc_cidr
+  instance_tenancy = "default"
+  tags = {
+    Name = "task1_vpc"
+  }
+}
+
+# Internet Gateway For Public Setup
+resource "aws_internet_gateway" "task1_igw" {
+  vpc_id = aws_vpc.task1_vpc.id
+  tags = {
+    Name = "task1_igw"
+  }
+}
+
+########
+
+# EIP for Public Nat Gateway
+resource "aws_eip" "task1_eip_public_nat_gateway" {
+  vpc   = true
+  count = length(var.pub_web_subnets_cidr)
+  tags = {
+    Name = "task1_eip_public_nat_gateway-${count.index + 1}"
+  }
+}
+#######
+# Nat Gateway for Public AZ1 & AZ2
+resource "aws_nat_gateway" "task1_public_nat_gateway" {
+  count         = length(var.pub_web_subnets_cidr)
+  subnet_id     = aws_subnet.task1_public_web[count.index].id
```

`count` should be separated from the rest of the arguments. Please fix it here and everywhere else in the code
```suggestion
  count = length(var.pub_web_subnets_cidr)
  
  subnet_id     = aws_subnet.task1_public_web[count.index].id
```